backend/rest_api_service/main.py

```diff
@@ -61,4 +61,125 @@ async def get_all_participants(response: Response, db: Session = Depends(get_db)
+async def create_event_with_required_associations(event: str, response: Response, db: Session = Depends(get_db)):
+    try:
+        await crud.create_event_with_required_associations(event=event, db=db)
+    except Exception as e:
+        response.status_code = 500
+        return {"message": f"An error occurred: {e}"}
+
+
+@app.post("/event/participants/create/{participant}", tags=['Event'], status_code=status.HTTP_201_CREATED)
+async def create_event_participant(participant: str, response: Response, db: Session = Depends(get_db)):
+    try:
+        await crud.create_event_participant(event_participants=participant, db=db)
+    except Exception as e:
+        response.status_code = 500
+        return {"message": f"An error occurred: {e}"}
+
+
+@app.post("/event/{event_id}/category/create/{category_id}", tags=['Event', 'Category'],
+          status_code=status.HTTP_201_CREATED)
+async def create_event_category(category_id: int, event_id: int, response: Response, db: Session = Depends(get_db)):
+    try:
+        await crud.create_event_category(event_id=event_id, category_id=category_id, db=db)
+    except Exception as e:
+        response.status_code = 500
+        return {"message": f"An error occurred: {e}"}
+
+
+@app.post("/category/create/{category}", tags=['Category'], status_code=status.HTTP_201_CREATED)
+async def create_category(category: str, response: Response, db: Session = Depends(get_db)):
+    try:
+        await crud.create_category(category=category, db=db)
+    except Exception as e:
+        response.status_code = 500
+        return {"message": f"An error occurred: {e}"}
+
+
+@app.post("/event/{event_id}/update/{changed_data}", tags=['Event'], status_code=status.HTTP_200_OK)
+async def update_event(event_id: int, changed_data: str, response: Response, db: Session = Depends(get_db)):
+    try:
+        await crud.update_event(event_id=event_id, changed_data=changed_data, db=db)
+    except Exception as e:
+        response.status_code = 500
+        return {"message": f"An error occurred: {e}"}
+
+
+@app.post("/category/{category_id}/update/{changed_data}", tags=['Category'], status_code=status.HTTP_200_OK)
+async def update_category(category_id: int, changed_data: str, response: Response, db: Session = Depends(get_db)):
+    try:
+        await crud.update_category(category_id=category_id, changed_data=changed_data, db=db)
+    except Exception as e:
+        response.status_code = 500
+        return {"message": f"An error occurred: {e}"}
+
+
+@app.post("/event/{event_id}/delete", tags=['Event'], status_code=status.HTTP_200_OK)
+async def delete_event(event_id: int, response: Response, db: Session = Depends(get_db)):
+    try:
+        await crud.delete_event(event_id=event_id, db=db)
+    except Exception as e:
+        response.status_code = 500
+        return {"message": f"An error occurred: {e}"}
+
+
+@app.post("/category/{category_id}/delete", tags=['Category'], status_code=status.HTTP_200_OK)
```

Comment:
też zamień tutaj POST na DELETE plus trzeba dodać walidację, czy przypadkiem category nie jest połączona z eventem. Jeśli jest, to nie możemy usunąć takiej kategorii

Reply:
@GrzesS0 nadal aktualne